x/lockup/keeper/lock_test.go

```diff
@@ -287,6 +318,90 @@ func (suite *KeeperTestSuite) TestUnlock() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestUnlockMaturedLockInternalLogic() {
+
+	testCases := []struct {
+		name                                    string
+		coinsLocked, coinsBurned, coinsSentBack sdk.Coins
+		expectedFinalCoinsSentBack              sdk.Coins
+
+		expectedError bool
+	}{
+		{
+			name:                       "unlock lock with gamm shares",
+			coinsLocked:                sdk.NewCoins(sdk.NewCoin("gamm/pool/1", sdk.NewInt(100))),
+			coinsBurned:                sdk.NewCoins(),
+			coinsSentBack:              sdk.NewCoins(sdk.NewCoin("gamm/pool/1", sdk.NewInt(100))),
+			expectedFinalCoinsSentBack: sdk.NewCoins(sdk.NewCoin("gamm/pool/1", sdk.NewInt(100))),
+			expectedError:              false,
+		},
+		{
+			name:                       "unlock lock with cl shares",
+			coinsLocked:                sdk.NewCoins(sdk.NewCoin("cl/pool/1/1", sdk.NewInt(100))),
+			coinsBurned:                sdk.NewCoins(sdk.NewCoin("cl/pool/1/1", sdk.NewInt(100))),
+			coinsSentBack:              sdk.NewCoins(),
+			expectedFinalCoinsSentBack: sdk.NewCoins(),
+			expectedError:              false,
+		},
```

Comment:
This should never be possible since cl shares never end up in a user account, so arbitrarily adding them to a lock would never happen. I still added it here as a defense in depth though! https://github.com/osmosis-labs/osmosis/pull/4884/commits/084c9fdcf13072ec39f3a2e3e2468a09e719fab0